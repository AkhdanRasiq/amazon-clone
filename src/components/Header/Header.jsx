import React from "react";

function Header() {
    return (
        <div>
            <p>yuhuuu</p>
        </div>
    )
}

export default Header
