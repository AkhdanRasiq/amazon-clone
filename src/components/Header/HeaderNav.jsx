import ShoppingBasketIcon from "@mui/icons-material/ShoppingBasket"

function HeaderNav() {
  return (
    <div className="headerNav">
      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Hello Guest</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Sign in</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Returns</p>
        </span>
        <span className="headerNavOptionItem">
          <p>& Orders</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Your</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Prime</p>
        </span>
      </div>

      <div className="headerNavOptionBasket">
        <ShoppingBasketIcon />
        <span className="headerNavOptionItem headerNavBasketCount">
          <p>0</p>
        </span>
      </div>
    </div>
  )
}

export default HeaderNav
