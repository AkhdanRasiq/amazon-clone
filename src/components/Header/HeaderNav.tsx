import React from 'react'

import ShoppingBasketIcon from "@mui/icons-material/ShoppingBasket"
import { Link } from 'react-router-dom'

function HeaderNav() {
  return (
    <div className="headerNav">
      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Hello Guest</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Sign in</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Returns</p>
        </span>
        <span className="headerNavOptionItem">
          <p>& Orders</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Your</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Prime</p>
        </span>
      </div>

      <Link to="/checkout">
        <div className="headerNavOptionBasket">
          <ShoppingBasketIcon />
          <span className="headerNavOptionItem headerNavBasketCount">
            <p>0</p>
          </span>
        </div>
      </Link>
    </div>
  )
}

export default HeaderNav
