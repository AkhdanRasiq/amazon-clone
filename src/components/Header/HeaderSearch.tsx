import React from 'react'

import SearchIcon from '@mui/icons-material/Search'

function HeaderSearch() {
    return (
        <div className="headerSearch">
            <input className="headerSearchInput" type="text" />
            <SearchIcon className="headerSearchButton" />
        </div>
    )
}

export default HeaderSearch
