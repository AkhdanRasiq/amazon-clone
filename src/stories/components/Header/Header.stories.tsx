import Header from '../../../components/Header/Header'
import '../../../assets/scss/styles.scss'

import StoryRouter from 'storybook-react-router'


export default {
  title: 'Components/Header',
  component: Header,
  // decorators: [StoryRouter()],
  argTypes: {
    submit: { action: 'submitted' }
  }
}

const Template = function (args: any, { argTypes }: any) {
  return (
      <div>
        <Header />
      </div>
  )
}

export const Default: any = Template.bind({})
