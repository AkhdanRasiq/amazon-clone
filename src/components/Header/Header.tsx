import React from 'react'

import HeaderSearch from './HeaderSearch'
import HeaderNav from './HeaderNav'

function Header() {
  return (
    <div className='header'>
      <img className="headerLogo" src={require('../../assets/img/logo_amazon.png')} alt="logo_amazon" />

      <HeaderSearch />

      <HeaderNav />
    </div>
  )
}

export default Header
