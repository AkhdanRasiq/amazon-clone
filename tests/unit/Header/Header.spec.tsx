import React from 'react'
import { render } from "@testing-library/react"

import Header from "../../../src/components/Header/Header"

describe("Header.tsx", () => {
  it("Render Properly", async () => {
    const wrapper = render(
      <Header />
    )
    await expect(wrapper).toMatchSnapshot()
  })
})
