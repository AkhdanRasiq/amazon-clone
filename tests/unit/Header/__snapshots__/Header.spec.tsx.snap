// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.tsx Render Properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <img
          alt="logo_amazon"
          class="headerLogo"
          src="[object Object]"
        />
        <div
          class="headerSearch"
        >
          <input
            class="headerSearchInput"
            type="text"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium headerSearchButton css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          class="headerNav"
        >
          <div
            class="headerNavOption"
          >
            <span
              class="headerNavOptionTitle"
            >
              <p>
                Hello Guest
              </p>
            </span>
            <span
              class="headerNavOptionItem"
            >
              <p>
                Sign in
              </p>
            </span>
          </div>
          <div
            class="headerNavOption"
          >
            <span
              class="headerNavOptionTitle"
            >
              <p>
                Returns
              </p>
            </span>
            <span
              class="headerNavOptionItem"
            >
              <p>
                & Orders
              </p>
            </span>
          </div>
          <div
            class="headerNavOption"
          >
            <span
              class="headerNavOptionTitle"
            >
              <p>
                Your
              </p>
            </span>
            <span
              class="headerNavOptionItem"
            >
              <p>
                Prime
              </p>
            </span>
          </div>
          <div
            class="headerNavOptionBasket"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingBasketIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
            <span
              class="headerNavOptionItem headerNavBasketCount"
            >
              <p>
                0
              </p>
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <img
        alt="logo_amazon"
        class="headerLogo"
        src="[object Object]"
      />
      <div
        class="headerSearch"
      >
        <input
          class="headerSearchInput"
          type="text"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium headerSearchButton css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        class="headerNav"
      >
        <div
          class="headerNavOption"
        >
          <span
            class="headerNavOptionTitle"
          >
            <p>
              Hello Guest
            </p>
          </span>
          <span
            class="headerNavOptionItem"
          >
            <p>
              Sign in
            </p>
          </span>
        </div>
        <div
          class="headerNavOption"
        >
          <span
            class="headerNavOptionTitle"
          >
            <p>
              Returns
            </p>
          </span>
          <span
            class="headerNavOptionItem"
          >
            <p>
              & Orders
            </p>
          </span>
        </div>
        <div
          class="headerNavOption"
        >
          <span
            class="headerNavOptionTitle"
          >
            <p>
              Your
            </p>
          </span>
          <span
            class="headerNavOptionItem"
          >
            <p>
              Prime
            </p>
          </span>
        </div>
        <div
          class="headerNavOptionBasket"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ShoppingBasketIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
            />
          </svg>
          <span
            class="headerNavOptionItem headerNavBasketCount"
          >
            <p>
              0
            </p>
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
